taxonomy.html

```diff
@@ -0,0 +1,478 @@
+          <code>visited</code> state.
+        </p>
+      </section>
+      <section class="informative">
+        <h3>
+          Understanding the terms.
+        </h3>
+        <p>
+          Each <a>accessible element</a> is described by accessible properties
+          like <a>role</a>, <a>state</a>, <a>actions</a> etc. Every accessible
+          property may take a range of values, for example,
+          <code>checkbox</code> and <code>paragraph</code> as values for
+          <a>role</a>. All accessible properties may have various dependencies,
+          for example, <code>checked</code> state can be applied to
+          <code>checkbox</code> role, however it's not applicable to
+          <code>paragraph</code> role. Accessible property values and their
+          dependencies are structured into <a>taxonomies</a>. The <a>assistive
+          technology</a> can operate with them.
+        </p>
+        <p>
+          The values of accessible properties and their relations are specific
+          to a markup whose semantics they serve to express. If the web author
+          needs to expose the semantics of some new kind, for example, to
+          describe a music sheet, then he may need to introduce new roles,
+          states etc and describe them. To do so the web author needs to extend
+          existing <a>taxonomies</a>. The <a>assistive technology</a> will able
+          then to understand the semantics of the content and figure out a
+          connection between 'known' taxa. For example, if the web author
+          introduces <code>checklistitem</code> role which is compound from two
+          <code>checkbox</code> and <code>listitem</code> roles, i.e it
+          inherits properties of both of them then the web author should
+          integrate the new role into existing role hierarchy.
+        </p>
+      </section>
+    </section>
+    <section>
+      <h2>
+        Extension to the <code>AccessibleDocument</code> interface
+      </h2>
+      <p>
+        An <a>accessible document</a> provides a method to import taxonomies.
+      </p>
+      <pre class="idl">
+partial interface AccessibleDocument {
+  Taxon? taxonOf(DOMString taxonomy, DOMString name);
+  [ Throws ]
+  void import(TaxoLibrary library);
+};
+</pre>
+      <section>
+        <h3>
+          <code>taxonOf</code> method
+        </h3>
+        <p>
+          Returns a taxon object of <code>Taxon</code> interface for the given
+          taxonomy and taxon name. Depending on the taxonomy the object may
+          implement other interfaces derived from <code>Taxon</code> interface.
+          For example, the returned object for the <code>role</code> taxonomy
+          implements <code>RoleTaxon</code> interface.
+        </p>
+        <p>
+          The example logs out all parents of the <code>bibliography</code>
+          role.
+        </p>
+        <pre class="example highlight">
+function logParents(taxon) {
```

Comment:
like this?
function logRoleTree(taxon, level) {
  for (let i = 0; i < taxon.children.length; i++) {
    var child = taxon.children[i];
    console.log(`{$level} ${child.name}`);
    logRoleTree(child, `${level}.${i}`);
  }
}
let taxon = document.a11ement.taxonOf('role', 'base');
logRoleTree(taxon);